pkg/util/errors/errors.go

```diff
@@ -4,17 +4,44 @@ import (
 	"net"
 	"net/url"
 	"os"
+	"strings"
 	"syscall"
 
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
 )
 
+// IsNetHTTPConnectionBrokenWrapperError returns true when we get the following error
+// net/http: HTTP/1.x transport connection broken
+func IsNetHTTPConnectionBrokenWrapperError(err error) bool {
+	return strings.Contains(err.Error(), "net/http: HTTP/1.x transport connection broken:")
+}
+
+// some common network errors that we check for
+var str2Syscall = map[string]syscall.Errno{
+	"connection reset by peer":  syscall.ECONNRESET,
+	"connection refused":        syscall.ECONNREFUSED,
+	"network is unreachable":    syscall.ENETUNREACH,
+	"no such file or directory": syscall.ENOENT,
+}
+
+func unwrapNetHTTPError(err error) (bool, error) {
```

Comment:
Any reason why `IsNetHTTPConnectionBrokenWrapperError` and `unwrapNetHTTPError` should not be in the same function? Considering they are very interconnected in the logic below? Or make `IsNetHTTPConnectionBrokenWrapperError` unexposed? as alone is not much of the value?

Reply:
I can simplify it, it was just how I started off ... 
I'll get rid of IsNetHTTPConnectionBrokenWrapperError() and just add the check for net/http in the unwrap function.